lib/babe/verify.go

```diff
@@ -330,13 +330,50 @@ func (b *verifier) verifyAuthorshipRight(header *types.Header) error {
 	if err != nil {
 		return fmt.Errorf("could not verify block equivocation: %w", err)
 	}
+
 	if equivocated {
 		return fmt.Errorf("%w for block header %s", ErrProducerEquivocated, header.Hash())
 	}
 
 	return nil
 }
 
+func (b *verifier) submitAndReportEquivocation(
+	slot uint64, authorityIndex uint32, firstHeader, secondHeader types.Header) error {
+
+	// TODO: Check if it is initial sync
+	// don't report any equivocations during initial sync
+	// as they are most likely stale.
+	// Issue #3004
+
+	bestBlockHash := b.blockState.BestBlockHash()
+	runtimeInstance, err := b.blockState.GetRuntime(bestBlockHash)
+	if err != nil {
+		return fmt.Errorf("getting runtime: %w", err)
+	}
+
+	offenderPublicKey := b.authorities[authorityIndex].ToRaw().Key
```

Comment:
what if `authorityIndex` is out of bounds? Maybe we should return an error in this case? Better than a panic.